libs/application/template-api-modules/src/lib/modules/templates/new-primary-school/new-primary-school.utils.ts

```diff
@@ -106,14 +128,83 @@ export const transformApplicationToNewPrimarySchoolDTO = (
           }
         : {}),
     },
+    ...(applicationType === ApplicationType.ENROLLMENT_IN_PRIMARY_SCHOOL
+      ? {
+          health: {
+            ...(hasFoodAllergiesOrIntolerances?.includes(YES)
+              ? {
+                  foodAllergiesOrIntolerances: foodAllergiesOrIntolerances,
+                }
+              : {}),
+            ...(hasOtherAllergies?.includes(YES)
+              ? {
+                  allergies: otherAllergies,
+                }
+              : {}),
+            ...(hasFoodAllergiesOrIntolerances?.includes(YES) ||
+            hasOtherAllergies?.includes(YES)
+              ? {
+                  usesEpipen: usesEpiPen === YES,
+                }
+              : {}),
+            hasConfirmedMedicalDiagnoses: hasConfirmedMedicalDiagnoses === YES,
+            requestsMedicationAdministration:
+              requestsMedicationAdministration === YES,
+          },
+        }
+      : {}),
     social: {
-      hasHadSupport: specialSupport === YES,
-      hasDiagnoses: developmentalAssessment === YES,
-    }, // Languages needs to be updated when Juni is ready with the data struccture
+      hasHadSupport: hasHadSupport === YES,
+      hasDiagnoses: hasDiagnoses === YES,
+      ...(hasHadSupport === YES || hasDiagnoses === YES
+        ? {
+            hasIntegratedServices: hasIntegratedServices === YES,
+            ...(hasIntegratedServices === YES
+              ? {
+                  hasCaseManager: hasCaseManager === YES,
+                  ...(hasCaseManager === YES
+                    ? {
+                        caseManager: {
+                          name: caseManagerName,
+                          email: caseManagerEmail,
+                        },
+                      }
+                    : {}),
+                }
+              : {}),
+          }
+        : {}),
+    },
     language: {
-      nativeLanguage: '',
-      noIcelandic: false,
-      otherLanguages: undefined,
+      languageEnvironment: languageEnvironment,
+      signLanguage: signLanguage === YES,
+      ...(languageEnvironment !== LanguageEnvironmentOptions.ONLY_ICELANDIC
+        ? {
+            preferredLanguage: preferredLanguage,
```

Comment:
Small nitpick. Can use object property shorthand
```suggestion
      languageEnvironment,
      signLanguage: signLanguage === YES,
      ...(languageEnvironment !== LanguageEnvironmentOptions.ONLY_ICELANDIC
        ? {
            preferredLanguage,
```

Reply:
Fixed 💯